src/jit/lower.cpp

```diff
@@ -2243,16 +2239,16 @@ GenTree* Lowering::LowerTailCallViaHelper(GenTreeCall* call, GenTree* callTarget
 #ifdef DEBUG
     argEntry = comp->gtArgEntryByArgNum(call, 0);
     assert(argEntry != nullptr);
-    assert(argEntry->node->gtOper == GT_PUTARG_REG);
-    GenTree* firstArg = argEntry->node->gtOp.gtOp1;
+    assert(argEntry->GetNode()->OperIs(GT_PUTARG_REG));
+    GenTree* firstArg = argEntry->GetNode()->AsUnOp()->gtGetOp1();
```

Comment:
Why do you need `>AsUnOp()` cast here and below? gtGetOp1() always exists. for the second we have `gtGetOp2IfPresent`.

Reply:
My crystal ball is telling me that these gtGetOp1/2 that are available on GenTree rather than GenTree(Un)Op are going to be a liability sooner or later. As such I tend to cast to the proper GenTree type when I add new code or when I change a line of existing code like in this case.